source/autocomplete.txt

```diff
@@ -6,10 +6,88 @@ Autocomplete
\ No newline at end of file
+The {+intellij-full+} provides autocomplete suggestions for database, 
+collection, or field names based on the connected data source. For example, 
+when you type a field name in a query, the plugin analyzes the MongoDB 
+collection or document schema to automatically suggest valid field names. 
+
+To use the autocomplete feature, start typing in a document field, collection, 
+or database name. Then press :kbd:`Ctrl` + :kbd:`Space` to display a drop-down 
+list that shows the available suggestions.
+
+Database Name Autocompletion
+----------------------------
+
+To autocomplete database names, press :kbd:`Ctrl` + :kbd:`Space` 
+inside the `getDatabase() <https://mongodb.github.io/mongo-java-driver/5.2/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html#getDatabase(java.lang.String)>`__ 
+method of a ``MongoClient`` instance.
+
+The {+intellij-short+} displays a drop-down list of available database names, 
+denoted by the ``MongoDB Database`` text and the database icon, as shown in 
+the image below: 
+
+.. image:: /images/database-autocomplete.png
+   :alt: Autocomplete database names in the IntelliJ Plugin.
+
+Collection Name Autocompletion
+------------------------------
+
+To autocomplete collection names, press :kbd:`Ctrl` + :kbd:`Space` 
+inside the `getCollection() <https://mongodb.github.io/mongo-java-driver/5.2/apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String)>`__ 
+method of a ``MongoClient`` instance.
+
+The {+intellij-short+} displays a drop-down list of available collection names, 
+denoted by the ``MongoDB Collection`` text and the collection icon, as shown in 
+the image below: 
+
+.. image:: /images/collection-autocomplete.png
+   :alt: Autocomplete collections names in the IntelliJ Plugin.
+
+.. important:: 
+
+   Collection name autocompletion currently requires a complete query to 
+   suggest collection names. For example, you must write a query that specifies 
+   the database name and a document field for the {+intellij-short+} to suggest 
+   valid collection names.
+
+Document Field Name Autocompletion 
+----------------------------------
+
+To autocomplete document field names, press :kbd:`Ctrl` + :kbd:`Space` 
+inside one of the following `MongoCollection <https://mongodb.github.io/mongo-java-driver/5.1/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html>`__ 
+methods: 
+
+- ``countDocuments``
+- ``deleteMany``
+- ``deleteOne``
+- ``distinct``
+- ``find``
+- ``findOneAndDelete``
+- ``findOneAndReplace``
+- ``findOneAndUpdate``
+- ``replaceOne``
+- ``updateMany``
+- ``updateOne``
+
+The {+intellij-short+} displays a drop-down list of available field names, 
+denoted by their data type and the document icon. For example, the image 
+below shows autocomplete suggestions for fields that start with ``dis`` in the 
+``trips.production`` collection: 
+
+.. image:: /images/document-autocomplete.png
```

Comment:
[Non-blocking Suggestion]:

The IDE is showing an error on the file and the line with the context menu due to the lack of a semicolon. While this does reflect real usage, (i.e., the editor automatically closing the parentheses without adding an EOL character, generating the error itself), seeing the red lines tweaked me enough to look for the cause instead of reading the example.

Reply:
Ah gotcha! Thanks for catching that. I'll update this.